File: dongyang_algo/day13/GetPrimeNumber.java
package day13;

import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

public class GetPrimeNumber {
    public static void main(String[] args) {
        System.out.println(getPrimeNumbers(3, 16));
    }

    public static List<Integer> getPrimeNumbers(int start, int end) {
        return IntStream.range(start, end).boxed()
                .filter(i -> IntStream.rangeClosed(2, (int) Math.sqrt(i))
                        .noneMatch(n -> (i % n == 0)))
                .collect(Collectors.toList());
    }
}

File: dongyang_algo/day13/GetPrimeNumber.py
import math

def is_prime_num(start, end):

    arr = [False, False] + [True] * (end - 1)

    for i in range(2, int(math.sqrt(end))+1):
        if arr[i]:
            j = 2
            while i * j <= end:
                arr[i*j] = False
                j += 1
    return [i for i in range(start, end+1) if arr[i]]


print(is_prime_num(3, 16))